code/game/objects/effects/aliens.dm

```diff
@@ -196,21 +196,20 @@
 		qdel(src)
 		return
 
-	direction_loop:
-		for(var/dirn in cardinal)
-			var/turf/T = get_step(src, dirn)
+	for(var/dirn in cardinal)
+		var/turf/T = get_step(src, dirn)
 
-			if (!istype(T) || T.density || locate(/obj/structure/alien/weeds) in T || istype(T, /turf/space))
-				continue
+		if (!istype(T) || T.density || locate(/obj/structure/alien/weeds) in T || istype(T, /turf/space))
+			continue
 
-			if(!linked_node || get_dist(linked_node, src) > linked_node.node_range)
-				return
+		if(!linked_node || get_dist(linked_node, src) > linked_node.node_range)
+			return
 
-			for(var/obj/O in T)
-				if(O.density)
-					continue direction_loop
+		var/turf/t_loc = loc
+		if(!t_loc.CanAtmosPass(T))
+			continue

```

Comment:
Nope nope nope nope nope.

This can be kinda costly to run, since it has to loop thru the contents of the turf and call a proc.

Why not check atmos_adjacent_turfs of the loc with the current direction, This is updated when a turf's atmos spread ability changes using CanAtmosPass(T), and this is what linda uses.

Or better yet, use the proc i made for scrubbers to do that: https://github.com/tgstation/-tg-station/blob/master/code/LINDA/LINDA_system.dm#L85 /turf/proc/GetAtmosAdjacentTurfs() returns a list of turfs that can share atmos with that one.